### PR TITLE
Proxy Protocol support

### PR DESCRIPTION
**Pull request by y-tabata**
Closes https://github.com/3scale/APIcast/issues/1194.

I have no idea for the prove test of the proxy protocol. I would be very happy if you give me any kind of comment.

### CODE REVIEW

File: doc/parameters.md [hunk 1]
```diff
@@ -424,6 +424,22 @@ Defines a HTTP proxy to be used for connecting to HTTPS services. Authentication
 
 Defines a comma-separated list of hostnames and domain names for which the requests should not be proxied. Setting to a single `*` character, which matches all hosts, effectively disables the proxy.
 
+### `APICAST_HTTP_PROXY_PROTOCOL`
+
+**Default:** false
+**Values:** boolean  
+**Example:** "true"
+
+Enables Proxy Protocol for the HTTP listener.
```
Comment (dfennessy) [inline doc/parameters.md:433]:
```suggestion
This parameter enables the Proxy Protocol for the HTTP listener.
```

File: doc/parameters.md [hunk 2]
```diff
@@ -424,6 +424,22 @@ Defines a HTTP proxy to be used for connecting to HTTPS services. Authentication
...
+
+### `APICAST_HTTPS_PROXY_PROTOCOL`
+
+**Default:** false
+**Values:** boolean  
+**Example:** "true"
+
+Enables Proxy Protocol for the HTTPS listener.
```
Comment (dfennessy) [inline doc/parameters.md:441]:
```suggestion
This parameter enables the Proxy Protocol for the HTTPS listener.
```

**Comment by y-tabata** [comment]
@dfennessy Thanks. I updated them.

**Comment by eloycoto** [comment]
This PR looks good to me, but I need to explore a good way to test this. It's not as straightforward as looks like, and I would like to have an integration coverage. 

I'll keep you posted. 

Thanks for your contribution!

**Comment by eloycoto** [comment]
made this one to fix this: https://github.com/3scale/APIcast/pull/1211

**Pull request closed without merging**